tasks/au/data.py

```diff
@@ -193,63 +194,15 @@ def columns(self):
                 id_ = line[0]               # A: Sequential
                 tablename = line[4]         # H: Tablename
-
                 # ignore tables we don't care about right now
                 if not id_.startswith(self.profile[0]) or \
                    not tablename.startswith(self.tablename):
                     continue
+                col_id = line[1]
+                rows[col_id] = line
 
-                col_id = line[1]            # B: short
-                col_name = line[2]          # C: name
-                denominators = line[3]      # D: denominators
-                col_unit = line[5]          # F: unit
-                col_subsections = line[6]   # G: subsection
-                if tablename == '{}02'.format(self.profile[0]):
-                    col_agg = line[8]       # I: AGG (for B02 only)
-                else:
-                    col_agg = None
-                tabledesc = line[10]        # K: Table description
-
-                denominators = denominators.split('|')
-
-                targets_dict = {}
-                for denom_id in denominators:
-                    denom_id = denom_id.strip()
-                    if not denom_id:
-                        continue
-
-                    reltype = 'denominator'
-                    if col_agg in ['median', 'average']:
-                        reltype = 'universe'
-
-                    if denom_id in column_reqs:
-                        targets_dict[column_reqs[denom_id].get(session)] = reltype
-                    else:
-                        targets_dict[cols[denom_id]] = reltype
-
-                targets_dict.pop(None, None)
-
-                cols[col_id] = OBSColumn(
-                    id=col_id,
-                    type='Numeric',
-                    name=col_name,
-                    description=tabledesc,
-                    # Ranking of importance, sometimes used to favor certain measures in auto-selection
-                    # Weight of 0 will hide this column from the user.  We generally use between 0 and 10
-                    weight=5,
-                    aggregate=col_agg or 'sum',
-                    # Tags are our way of noting aspects of this measure like its unit, the country
-                    # it's relevant to, and which section(s) of the catalog it should appear in
-                    tags=[source, license, country, unittags[col_unit]],
-                    targets=targets_dict
-                )
-
-                # append the rest of the subsection tags
-                col_subsections = col_subsections.split('|')
-                for subsection in col_subsections:
-                    subsection = subsection.strip()
-                    subsection_tag = subsectiontags[subsection]
-                    cols[col_id].tags.append(subsection_tag)
+        for col_id in rows:
+            self._process_col(rows[col_id], rows, session, column_reqs, cols, tablename, source, license, country, unittags, subsectiontags)
```

Comment:
It feels weird to me to pass both `rows[col_id]` and `rows` to this function. I'd either:
a) Pass `col_id` and `rows`
b) Pass `rows` and do the loop inside the function

Might be just me, I'm known for having crazy criteria for these things :P

Also, `license` is a python built-in, best to avoid using it as a variable name if possible.